packages/components/table/src/store/expand.ts

```diff
@@ -27,7 +27,7 @@ function useExpand<T>(watcherData: WatcherPropsData<T>) {
         return prev
       }, [])
     } else {
-      expandRows.value = []
```

Comment:
this line of code is to reset the expand status when table rerender.
like this gif:
![table-expand](https://user-images.githubusercontent.com/30046649/225278577-1cbc7cfd-bb9d-4fae-b155-b06759573a0c.gif)

so, If we change the reactive data when we open the expand rows that row open will fail.
like this [playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5sZXQgY2xpY2tFeHBhbmQgPSByZWYoMClcbmNvbnN0IHRhYmxlRGF0YSA9cmVmKFtcbiAge1xuICAgIG5hbWU6IGBUT00tJHtjbGlja0V4cGFuZC52YWx1ZX1gLFxuICB9LFxuXSlcblxuY29uc3QgdGFibGVFeHBhbmRIYW5kbGUgPSAoaXRlbSk9PntcbiAgY2xpY2tFeHBhbmQudmFsdWUgKz0gMVxuICBpdGVtLmEgPSBNYXRoLnJhbmRvbSgpXG59XG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8cD5jbGlja0V4cGFuZCA6IHt7Y2xpY2tFeHBhbmR9fTwvcD5cbiAgPGVsLXRhYmxlIDpkYXRhPVwidGFibGVEYXRhXCIgQGV4cGFuZC1jaGFuZ2U9XCJ0YWJsZUV4cGFuZEhhbmRsZVwiICA6Ym9yZGVyPVwidHJ1ZVwiPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gdHlwZT1cImV4cGFuZFwiPlxuICAgICAgPHRlbXBsYXRlICNkZWZhdWx0PVwicHJvcHNcIj5cbiAgICAgICAgMTIzXG4gICAgICA8L3RlbXBsYXRlPlxuICAgIDwvZWwtdGFibGUtY29sdW1uPlxuICAgIDxlbC10YWJsZS1jb2x1bW4gbGFiZWw9XCJOYW1lXCIgcHJvcD1cIm5hbWVcIiAvPlxuICA8L2VsLXRhYmxlPlxuXG4gIDxwPlxuICAgIDxlbC1pY29uIGNvbG9yPVwidmFyKC0tZWwtY29sb3ItcHJpbWFyeSlcIj48RWxlbWVudFBsdXMgLz48L2VsLWljb24+XG4gICAgRWxlbWVudCBQbHVzIHt7IEVwVmVyc2lvbiB9fSArIFZ1ZSB7eyB2dWVWZXJzaW9uIH19XG4gIDwvcD5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)